### PR TITLE
Change TeamAcceptInviteOrRequestAccess to return more info

### PR DESCRIPTION
**Pull request by zapu**

### CODE REVIEW

**Comment by zapu** [comment]
cc @buoyad

File: go/teams/service_helper.go [hunk 1]
```diff
@@ -835,14 +835,25 @@ func RequestAccess(ctx context.Context, g *libkb.GlobalContext, teamname string)
...
-func TeamAcceptInviteOrRequestAccess(ctx context.Context, g *libkb.GlobalContext, tokenOrName string) error {
+func TeamAcceptInviteOrRequestAccess(ctx context.Context, g *libkb.GlobalContext, tokenOrName string) (keybase1.TeamAcceptOrRequestResult, error) {
 	// First try to accept as an invite
 	err := AcceptInvite(ctx, g, tokenOrName)
-	if err != nil {
-		// Failing that, request access as a team name
-		_, err = RequestAccess(ctx, g, tokenOrName)
+	if err == nil {
+		return keybase1.TeamAcceptOrRequestResult{
+			WasToken: true,
```
Comment (mlsteele) [inline go/teams/service_helper.go]: What does `Open` mean? If it's whether the team is open, then it's missing here. If it's whether the request that just sent was an open team request then ok but it deserves a comment or name change

Comment (zapu) [reply go/teams/service_helper.go]: It's used to indicate whether RequestAccess(teamname) had to send request to team admins that they have to approve manually; or if it's an open team and user will be added automatically. What would be a better name for that?

Comment (mlsteele) [reply go/teams/service_helper.go]: Could be `WasOpenRequest`. But you could just throw that comment in avdl
``
// Whether a request was sent to team admins that they will automatically approve.
```

**Comment by zapu** [comment]
I'll merge this after trying to launch gui from this branch to see if nothing broke

**Comment by mlsteele** [comment]
I'm going to borrow this branch

**Comment by mlsteele** [comment]
Changes:
- downcase seitan tokens (ikeys) on the way in.
- Change error handling to allow requesting access to 16 character teams. Which now unlocks further unpleasantness of sending the ikey up.

File: go/teams/service_helper.go [hunk 2]
```diff
@@ -663,6 +663,8 @@ func AcceptSeitan(ctx context.Context, g *libkb.GlobalContext, ikey SeitanIKey)
 		return err
 	}
 
+	g.Log.CDebugf(ctx, "seitan invite ID: %v", inviteID)
```
Comment (maxtaco) [inline go/teams/service_helper.go]: are we sure this is OK to output?

Comment (maxtaco) [reply go/teams/service_helper.go]: i guess so...

**Comment by zapu** [comment]
@mlsteele good catch about 16-character teams. thanks

**Pull request merged**